From 7ef2ca65257a025dfdc7a7dee3a1d135ac14dde7 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Wed, 06 Jul 2022 17:30:57 +0300
Subject: [PATCH] [SW-94367] gaudi2: change sris mode programming sequence

This patch changes SRIS mode programming sequence as below recommended
by arch team.
1. Write 1 to PCIE_PHY_PIPE_RX_SRIS_MODE_EN
2. Write 0 to PCIE_PHY_PIPE_LANE_CMN_REFCLK_MODE
3. Write 0 to PCIE_PHY_PHYX_MPLL_SHORT_LOCK_EN_0/1/2/3
4. Write 1 to PCIE_AUX_APP_SRIS_MODE

Change-Id: Ic8c68623f1c7cf0270fdbe48275cc673b99fd11a
---

diff --git a/common/pcie.c b/common/pcie.c
index 6e51efe..ec0f752 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -97,8 +97,8 @@
 		pcie_phy_para_sel_set();
 
 		if (get_sris_mode_from_strap()) {
-			pcie_phy_set_lane_refclk_mode();
 			regmap_writel(1, mmPIPE_RX_SRIS_MODE_EN);
+			pcie_phy_set_lane_refclk_mode();
 			pcie_phy_set_pll_short_lock();
 			regmap_writel(1, mmAPP_SRIS_MODE);
 		}
